src/live.rs

```diff
@@ -116,10 +141,34 @@ pub fn iso_ignition_show(config: &IsoIgnitionShowConfig) -> Result<()> {
 }
 
 pub fn iso_ignition_remove(config: &IsoIgnitionRemoveConfig) -> Result<()> {
-    let mut holder = CopiedFileHolder::new(&config.input, config.output.as_ref())?;
-    let mut embed = EmbedArea::for_file(&mut holder.file)?;
-    embed.clear()?;
-    holder.complete();
+    let use_stdout = config.output.as_deref() == Some("-");
+    if use_stdout && isatty(stdout().as_raw_fd()).chain_err(|| "checking if stdout is a TTY")? {
+        bail!("Refusing to write binary data to terminal");
+    }
+
+    let mut holder: CopiedFileHolder;
+    let mut embed: EmbedArea;
+
+    if use_stdout {
+        holder = CopiedFileHolder {
+            file: OpenOptions::new()
+                .read(true)
+                .open(&config.input)
+                .chain_err(|| format!("opening {}", &config.input))?,
+            copied_path: None,
+            complete: false,
+        };
+    } else {
+        holder = CopiedFileHolder::new(&config.input, config.output.as_ref())?;
+    }
+    embed = EmbedArea::for_file(&mut holder.file)?;
+
```

Comment:
Since we don't have a meaningful set of common code between the two cases here, it might be cleaner to just have a single if/else block.